docs/src/man/comparisons.md

```diff
@@ -1,77 +1,248 @@
+                   'z': [3, 4, 5, 6, 7, None]},
+                   index = list('abcdef'))
+df2 = pd.DataFrame({'grp': [1, 3], 'w': [10, 11]})
+```
+
+Because pandas supports multi-index, this example data frame is set up with `a` to `f`
+as row indices rather than a separate `id` column.
+
+### Accessing data
+
+| Operation                  | pandas                 | DataFrames.jl                      |
+|----------------------------|------------------------|------------------------------------|
+| Cell indexing by location  | `df.iloc[1, 1]`        | `df[2, 2]`                         |
+| Row slicing by location    | `df.iloc[1:3]`         | `df[2:3, :]`                       |
+| Column slicing by location | `df.iloc[:, 1:]`       | `df[:, 2:end]`                     |
+| Row indexing by label      | `df.loc['c']`          | `df[findfirst(==('c'), df.id), :]` |
+| Column indexing by label   | `df.loc[:, 'x']`       | `df[:, :x]`                        |
+| Column slicing by label    | `df.loc[:, ['x','z']]` | `df[:, [:x, :z]]`                  |
+|                            | `df.loc[:, 'x':'z']`   | `df[:, Between(:x, :z)]`           |
+| Mixed indexing             | `df.loc['c'][1]`       | `df[findfirst(==('c'), df.id), 2]` |
+
+Note that Julia uses 1-based indexing, inclusive on both ends. A special keyword `end` can be used to
+indicate the last index.
+
+In addition, the `findfirst` function is used to find the first match and return the result
+as a single `DataFrameRow` object. In the case that `id` is not unique, you can use the `findall` function
+or boolean indexing instead. It would then return a `DataFrame` object containing all matched rows. The following
+two lines of code are functionally equivalent:
+
+```julia
+df[findall(==('c'), df.id), :]
+df[df.id .== 'c', :]
+```
+
+DataFrames.jl's indexing always produces a consistent and predictable return type.
+By contrast, pandas' `loc` function returns a `Series` object when there is exactly
+one `'c'` value in the index, and it returns a `DataFrame` object when there are multiple
+rows having the index value of `'c'`.
+
+### Common operations
+
+| Operation                | pandas                                                         | DataFrames.jl                               |
+|--------------------------|----------------------------------------------------------------|---------------------------------------------|
+| Reduce multiple values   | `df['z'].mean(skipna = False)`                                 | `mean(df.z)`                                |
+|                          | `df['z'].mean()`                                               | `mean(skipmissing(df.z))`                   |
+|                          | `df[['z']].agg(['mean'])`                                      | `combine(df, :z => mean ∘ skipmissing)`     |
+| Add new columns          | `df.assign(z1 = df['z'] + 1)`                                  | `transform(df, :z => (v -> v .+ 1) => :z1)` |
```

Comment:
I would show three options here as `transform` in this case is actually a most complex one:
```suggestion
| Add new columns          | `df.assign(z1 = df['z'] + 1)`                                  | `df.z1 = df.z .+ 1`                         |
|                          |                                                                | `insertcols!(df, :x1 => df.z .+ 1)`         |
|                          |                                                                | `transform(df, :z => (v -> v .+ 1) => :z1)` |
```

Reply:
Done with minor edit (using `:z1` rather than `:x1` for `insertcols!`). See commit https://github.com/JuliaData/DataFrames.jl/pull/2378/commits/32b36b33531defe4af1f9fcdebbce7681899b1c8